src/test/java/use_case/execute_buy/ExecuteBuyInteractorTest.java

```diff
@@ -0,0 +1,98 @@
+package use_case.execute_buy;
+
+import entity.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class ExecuteBuyInteractorTest {
+
+    private ExecuteBuyDataAccessInterface dataAccess;
+    private ExecuteBuyOutputBoundary outputPresenter;
+    private UserFactory userFactory;
+
+    @BeforeEach
+    void setUp() {
+        userFactory = new UserFactory();
+        dataAccess = Mockito.mock(ExecuteBuyDataAccessInterface.class);
+        outputPresenter = Mockito.mock(ExecuteBuyOutputBoundary.class);
+    }
+
+    @Test
+    void successTest() throws ExecuteBuyDataAccessInterface.ValidationException {
+        User mockUser = userFactory.create("testUser", "password");
+        mockUser.addBalance(10000.0);
+
+        when(dataAccess.getUserWithCredential("00000")).thenReturn(mockUser);
+
+        Stock stock = new Stock("XXXX", 100.0);
+
+        try (MockedStatic<StockMarket> mockedStatic = Mockito.mockStatic(StockMarket.class)) {
+            // prepare StockMarket mock
+            StockMarket stockMarketMock = Mockito.mock(StockMarket.class);
+            mockedStatic.when(StockMarket::Instance).thenReturn(stockMarketMock);
+
+            // prepare StockMarket mock to return the stock
+            when(stockMarketMock.getStock("XXXX")).thenReturn(Optional.of(stock));
+
+            // prepare input data
+            ExecuteBuyInputData inputData = new ExecuteBuyInputData("00000", "XXXX", 100);
+
+            // create interactor
+            ExecuteBuyInteractor interactor = new ExecuteBuyInteractor(dataAccess, outputPresenter);
+            interactor.execute(inputData);
+
+            // check if message is prepared
+            verify(outputPresenter).prepareSuccessView(new ExecuteBuyOutputData("Purchase successful."));
+
+            // check if user portfolio contains the stock
+            assertTrue(mockUser.getPortfolio().getUserStock("XXXX").isPresent(), "Portfolio should contain the ticker XXXX");
+
+            // check if user stock quantity is correct
+            UserStock userStock = mockUser.getPortfolio().getUserStock("XXXX").get();
+            assertEquals(100, userStock.getQuantity(), "Portfolio should contain 100 shares of XXXX");
+        }
```

Comment:
For this test, we should also check whether the user's new balance is correctly updated